cpp/src/groupby/sort/group_nunique.cu

```diff
@@ -121,15 +84,33 @@ std::unique_ptr<column> group_nunique(column_view const& values,
   CUDF_EXPECTS(static_cast<size_t>(values.size()) == group_labels.size(),
                "Size of values column should be same as that of group labels");
 
-  return type_dispatcher(values.type(),
-                         nunique_functor{},
-                         values,
-                         group_labels,
-                         num_groups,
-                         group_offsets,
-                         null_handling,
-                         stream,
-                         mr);
+  auto result = make_numeric_column(
+    data_type(type_to_id<size_type>()), num_groups, mask_state::UNALLOCATED, stream, mr);
+
+  if (num_groups == 0) { return result; }
+
+  auto const values_view = table_view{{values}};
+  auto const comparator  = cudf::experimental::row::equality::self_comparator{values_view, stream};
+  auto const d_equal     = comparator.equal_to(
+    cudf::nullate::DYNAMIC{cudf::has_nested_nulls(values_view)}, null_equality::EQUAL);
+
+  auto const d_values_view      = column_device_view::create(values, stream);
+  auto const is_unique_iterator = thrust::make_transform_iterator(
+    thrust::make_counting_iterator(0),
```

Comment:
I notice that the old code was explicitly templating the type of this iterator, is that important to preserve for any reason?

Reply:
Replaced by `thrust::counting_iterator<cudf::size_type>(0)`